File: 2021/Day1/Day1.c
#include <stdio.h>
#include <stdlib.h>

void main() {
    FILE *file = fopen("Day1.txt", "r");
    char line[255];

    int depth[4];
    for (size_t i = 0; i < 3; i++) {
        depth[i] = atoi(fgets(line, 255, file));
    }

    int res = 0;
    while (fgets(line, 255, file)) {
        depth[3] = atoi(line);
        if (depth[3] > depth[0]) {
            res++;
        }
        for (size_t i = 0; i < 4; i++) {
            printf("%d ", depth[i]);
        }
        printf("\n");
        for (size_t i = 0; i < 3; i++) {
            depth[i] = depth[i + 1];
        }
    }
    printf("%d\n", res);
}

File: 2021/Day1/Day1.java
import java.io.*;
import java.util.Scanner;

public class Day1 {
    public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(new File("Day1.txt"));

        int[] depth = new int[4];
        for (int i = 0; i < 3; i++) {
            depth[i] = Integer.parseInt(scanner.nextLine());
        }

        int res = 0;
        while (scanner.hasNext()) {
            depth[3] = Integer.parseInt(scanner.nextLine());
            if (depth[3] > depth[0]) {
                res += 1;
            }
            for (int i = 0; i < 3; i++) {
                depth[i] = depth[i + 1];
            }
        }
        System.out.println(res);
    }
}